### PR TITLE
Sendinblue use case

### PR DESCRIPTION
**Pull request by tbedford**
## Description

Sendinblue use case.

## Code

* https://github.com/nexmo-community/sendinblue-use-case

## Review

* https://nexmo-developer-pr-2061.herokuapp.com/

### CODE REVIEW

File: _use_cases/client-sdk-sendinblue-order-confirm.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+This creates an order for user 'user-123'. For simplicity, this is a simple pre-defined static order, rather than a full-fledged shopping cart.
+
+This step also generates a custom event of type `custom:order-confirm-event` containing the order details.
+
+In addition, a confirmation email is sent via Sendinblue. This email contains a link the user would select to chat if they wanted support with order.
+
+**5.** Check you have received the order email. Go to the inbox defined in your configuration to read the confirmation email.
+
+**6.** Click the link in the email to log the customer into the chat screen.
```
Comment (marklewin) [inline _use_cases/client-sdk-sendinblue-order-confirm.md]: The link uses port 9000, whereas I configured mine to use 3000 ?

File: _use_cases/client-sdk-sendinblue-order-confirm.md [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+
+1. Creates a `NexmoClient` instance.
+2. Logs the user into the Conversation based on a JWT generated by the server.
+3. Obtains the Conversation object.
+4. Registers event handlers for the message send button and `text` events.
+5. Provides a basic UI for displaying current order, order history and message history, as well as the ongoing chat.
+
+## Summary
+
+In this use case you have learned how to build an order confirmation and support system. The user receives an oeder confirmation email via Sendinblue. The user can then engage in two-way messaging with the support agent to discuss the order if required.
```
Comment (marklewin) [inline _use_cases/client-sdk-sendinblue-order-confirm.md]:
```suggestion
In this use case you have learned how to build an order confirmation and support system. The user receives an order confirmation email via Sendinblue. The user can then engage in two-way messaging with the support agent to discuss the order if required.
```

File: _use_cases/client-sdk-sendinblue-order-confirm.md [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+
+Add in your application ID from the installation section. You can obtain your API key and secret from the [Nexmo Dashboard](https://dashboard.nexmo.com).
+
+The private key file will typically be `private.key`, unless you specified something else.
+
+> **NOTE:** The Conversation ID is only used for testing purposes. You do not need to configure it at this stage.
+
+### Sendinblue configuration
+
+You must have a [Sendinblue](https://www.sendinblue.com) API key.
```
[the remark this answers is not in the document]
Comment (tbedford) [reply _use_cases/client-sdk-sendinblue-order-confirm.md]: Fixed.

File: _use_cases/client-sdk-sendinblue-order-confirm.md [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+
+For testing this use case it is assumed you have Sendinblue "sender" information. This is the email address and name you are sending emails from. You will also want to specify a user name and email address that will receive the order confirmation emails. Usually this information would be available on a per-customer basis in the user database, but in this use case it is set in the environment file for testing convenience.
+
+You also need the ID of the email template you are using. The template is created in the Sendinblue UI. When you have created a template and activated it you can make a note of the ID as specified in the UI. This is the number that is used here.
```
[the remark this answers is not in the document]
Comment (tbedford) [reply _use_cases/client-sdk-sendinblue-order-confirm.md]: Fixed.

File: _use_cases/client-sdk-sendinblue-order-confirm.md [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+
+**6.** Copy `example.env` to `.env` in the project directory. You will edit `.env` in a later step to specify your credentials and other configuration information.
+
+**7.** Create a Nexmo application [interactively](/application/nexmo-cli#interactive-mode). The following command enters interactive mode:
+
+``` bash
+nexmo app:create
+```
+
+Specify your application name and also select RTC capabilities. A private key will be written out to `private.key`.
```
[the remark this answers is not in the document]
Comment (tbedford) [reply _use_cases/client-sdk-sendinblue-order-confirm.md]: I've reworked this section to give step-by-step instructions.

File: _use_cases/client-sdk-sendinblue-order-confirm.md [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+SENDINBLUE_FROM_NAME=
+SENDINBLUE_FROM_EMAIL=
+SENDINBLUE_TO_NAME=
+SENDINBLUE_TO_EMAIL=
+SENDINBLUE_TEMPLATE_ID=
+```
+
+This assumes you are using port 3000, but you can use any convenient free port.
+
+Add in your application ID from the installation section. You can obtain your API key and secret from the [Nexmo Dashboard](https://dashboard.nexmo.com).
```
[the remark this answers is not in the document]
Comment (tbedford) [reply _use_cases/client-sdk-sendinblue-order-confirm.md]: Added.

File: _use_cases/client-sdk-sendinblue-order-confirm.md [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+
+**1.** In the project directory start the server:
+
+``` bash
+npm start
+```
+
+This starts up the server using `node.js`.
+
+**2.** Create the support agent user with the following Curl command:
```
[the remark this answers is not in the document]
Comment (tbedford) [reply _use_cases/client-sdk-sendinblue-order-confirm.md]: Thank you. The `process.on` snippet was extremely useful! I have also added some console logging and referenced that from the docs.

File: _use_cases/client-sdk-sendinblue-order-confirm.md (hunk 1 above)
Comment (tbedford) [reply _use_cases/client-sdk-sendinblue-order-confirm.md]: Oops, yes, I'd inadvertently hard-coded the port number into the URL. Fixed.

File: _use_cases/client-sdk-sendinblue-order-confirm.md (hunk 2 above)
Comment (tbedford) [reply _use_cases/client-sdk-sendinblue-order-confirm.md]: Fixed typo.

File: _use_cases/client-sdk-sendinblue-order-confirm.md [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+
+3. A chat screen is loaded that contains the current order data, order history, and message history. The order and message histories are stored in a Conversation associated with the user.
+
+4. Two way chat can then take place between the customer and a support agent.
+
+## Installation
+
+The following procedure assumes you have the `git` and `npm` commands available on the command line.
+
+**1.** Install the Nexmo CLI:
```
[the remark this answers is not in the document]
Comment (tbedford) [reply _use_cases/client-sdk-sendinblue-order-confirm.md]: Yep. Fixed.

File: _use_cases/client-sdk-sendinblue-order-confirm.md [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+
+**7.** Create a Nexmo application [interactively](/application/nexmo-cli#interactive-mode). The following command enters interactive mode:
+
+``` bash
+nexmo app:create
+```
+
+Specify your application name and also select RTC capabilities. A private key will be written out to `private.key`.
+
+The file `.nexmo-app` is created in the project directory containing the Application ID and the private key. Make a note of the Application ID as you will need this later.
```
[the remark this answers is not in the document]
Comment (tbedford) [reply _use_cases/client-sdk-sendinblue-order-confirm.md]: Whole section tidied up to be more explicit.

File: _use_cases/client-sdk-sendinblue-order-confirm.md [hunk 10]
```diff
@@ -0,0 +1,206 @@
...
+SENDINBLUE_TO_NAME=
+SENDINBLUE_TO_EMAIL=
+SENDINBLUE_TEMPLATE_ID=
+```
+
+This assumes you are using port 3000, but you can use any convenient free port.
+
+Add in your application ID from the installation section. You can obtain your API key and secret from the [Nexmo Dashboard](https://dashboard.nexmo.com).
+
+The private key file will typically be `private.key`, unless you specified something else.
```
[the remark this answers is not in the document]
Comment (tbedford) [reply _use_cases/client-sdk-sendinblue-order-confirm.md]: Fixed.

**Pull request merged**